### ISSUE TITLE
mongodb synchronization - selecting commands

### ISSUE THREAD

**Issue by SamNakhaei** [alibaba/MongoShake#569]
Dear all,
I would like to do a synchronization of a MongoDB replica set (node-1) and another MongoDB (node-2) but only for update/insert operations. (direct tunnel)
so, when I add/update a document in collection A of node-1, I should see this change in node-2 as well. (which MongoShake does it perfectly). but, if a document gets removed from node-1, I don't want this removal to happen in node-2 as well (which MongoShake removes the document in node-2).
so, basically, I want MongoShake to perform "i", "u" and "n" but not "d"

do you have any suggestions on how it is possible with MongoShake?
Can I do this task just within the direct tunnel, or do I need to have my own service over mango shake (through other tunnels like kafka)?

Many thanks

**Comment by vinllen** [comment]
Why do you want to filter the delete operation?
Currently, MongoShake can not filter operation=delete('d') oplog which will lead to inconsistencies between the source and target mongodb.
If you really want that feature, you can use kafka tunnel to fetch all oplog and filter the op=d oplog, and then inject the data into the target MongoDB. However, this method needs some works.

**Comment by SamNakhaei** [comment]
the plan is to have a different storage policy on the nodes. since node-1 will have limited storage, I want to aggregate all the data on node-2. so, it would be like 1 day of data on node-1 and 6 months of data on node-2. The incremental option on MongoShake is super useful and good for this. but it also deletes the documents when they get deleted on node-1.

thanks for the advice on Kafka tunnel. is there any document that shows what mongoshake will provide and what's the structure?
